### PR TITLE
Reworks burning objects to be a component

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Title.

## Why It's Good For The Game

Simply put, allows for atoms which are not /obj but use atom_integrity to burn up too, which is nice and good.
But also, it allows for neat behavior like burning particle effects (only structures use that right now to spawn smoke)
image

## Changelog

:cl:
add: Burning structures spawn smoke particles. Sick.
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
i wanted the particles to apply to items too but it looked really really bad because then when worn the mob gets the particles and they stacked and it was terrible

File: code/game/objects/obj_defense.dm [hunk 1]
```diff
@@ -131,16 +132,9 @@ GLOBAL_DATUM_INIT(acid_overlay, /mutable_appearance, mutable_appearance('icons/e
 ///called when the obj is destroyed by fire
 /obj/proc/burn()
 	if(resistance_flags & ON_FIRE)
-		SSfire_burning.processing -= src
+		qdel(GetComponent(/datum/component/burning))
```
[the remark this answers is not in the document]
Comment (ChungusGamer666) [reply code/game/objects/obj_defense.dm:135]: ok gonna do what you said (this proc is dumb as hell tho)

[the remark this answers is not in the document]
Comment (ChungusGamer666) [reply code/game/objects/obj_defense.dm:135]: i'm just gonna remove this qdel since destroy already deals with this

**Comment by ChungusGamer666** [comment]
this is less shrimple than i thought

**Comment by ChungusGamer666** [comment]
ok this should be good to go yea

**Comment by dwasint** [comment]
> i wanted the particles to apply to items too but it looked really really bad because then when worn the mob gets the particles and they stacked and it was terrible

you need to have a dummy object holding the particles thats rendered on the item itself to not have the particles transfer over, this also allows you to have multiple particles slapped onto a single object

**Comment by LemonInTheDark** [comment]
we can fix him

**Comment by LemonInTheDark** [comment]
#74524 adds support for enabling/disabling the "smoke the parent" behavior via a flag, which defaults to off.
Would be trivial to remove the flag for items, and add it for like, mobs, so burning in a locker gives the locker smoke effects and shit.

**Pull request merged**